src/features/defi/providers/foxy/components/FoxyManager/Withdraw/FoxyWithdraw.tsx

```diff
@@ -0,0 +1,585 @@
+        title: translate('common.transactionFailed'),
+        status: 'error'
+      })
+    } finally {
+      dispatch({ type: FoxyWithdrawActionType.SET_LOADING, payload: false })
+    }
+  }
+
+  const handleConfirm = async () => {
+    try {
+      if (!state.userAddress || !rewardId || !walletState.wallet) return
+      dispatch({ type: FoxyWithdrawActionType.SET_LOADING, payload: true })
+      const [txid, gasPrice] = await Promise.all([
+        api.withdraw({
+          tokenContractAddress: rewardId,
+          userAddress: state.userAddress,
+          contractAddress,
+          wallet: walletState.wallet,
+          amountDesired: bnOrZero(state.withdraw.cryptoAmount)
+            .times(`1e+${asset.precision}`)
+            .decimalPlaces(0),
+          type: state.withdraw.withdrawType
+        }),
+        api.getGasPrice()
+      ])
+      dispatch({ type: FoxyWithdrawActionType.SET_TXID, payload: txid })
+      history.push(WithdrawPath.Status)
+
+      const transactionReceipt = await poll({
+        fn: () => api.getTxReceipt({ txid }),
+        validate: (result: TransactionReceipt) => !isNil(result),
+        interval: 15000,
+        maxAttempts: 30
+      })
+      dispatch({
+        type: FoxyWithdrawActionType.SET_WITHDRAW,
+        payload: {
+          txStatus: transactionReceipt.status === true ? 'success' : 'failed',
+          usedGasFee: bnOrZero(gasPrice).times(transactionReceipt.gasUsed).toFixed(0)
+        }
+      })
+      dispatch({ type: FoxyWithdrawActionType.SET_LOADING, payload: false })
+    } catch (error) {
+      console.error('FoxyWithdraw:handleConfirm error', error)
+    }
+  }
+
+  const handleViewPosition = () => {
+    browserHistory.push('/defi')
+  }
+
+  const handleCancel = () => {
+    browserHistory.goBack()
+  }
+
+  const validateCryptoAmount = (value: string) => {
+    const crypto = bnOrZero(balance).div(`1e+${asset.precision}`)
+    const _value = bnOrZero(value)
+    const hasValidBalance = crypto.gt(0) && _value.gt(0) && crypto.gte(value)
+    if (_value.isEqualTo(0)) return ''
+    return hasValidBalance || 'common.insufficientFunds'
+  }
+
+  const validateFiatAmount = (value: string) => {
+    const crypto = bnOrZero(balance).div(`1e+${asset.precision}`)
+    const fiat = crypto.times(marketData.price)
+    const _value = bnOrZero(value)
+    const hasValidBalance = fiat.gt(0) && _value.gt(0) && fiat.gte(value)
+    if (_value.isEqualTo(0)) return ''
+    return hasValidBalance || 'common.insufficientFunds'
+  }
+
+  const cryptoAmountAvailable = bnOrZero(balance).div(`1e+${asset?.precision}`)
+  const fiatAmountAvailable = bnOrZero(cryptoAmountAvailable).times(marketData.price)
+
+  const renderRoute = (route: { step?: number; path: string; label: string }) => {
+    let statusIcon: React.ReactElement = <ArrowForwardIcon />
+    let statusText = StatusTextEnum.pending
+    if (state.withdraw.txStatus === 'success') {
```

Comment:
Check out the changes I made to foxy deposit status stuff. We pass in a BG color now -- https://github.com/shapeshift/web/blob/380c1696dfc77132cdbc68ace55249542217881c/src/features/defi/providers/foxy/components/FoxyManager/Deposit/FoxyDeposit.tsx#L356